lib/ansible/modules/windows/win_mssql_login.ps1

```diff
@@ -0,0 +1,188 @@
+#!powershell
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Daniele Lazzari  <lazzari@mailup.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$ErrorActionPreference = "Stop"
+$result = @{
+    'changed' = $False;
+    'output'  = $null
+}
+
+Function Test-SQLUser {
+    Param(
+        [string]$LoginName,
+        [string]$ServerInstance,
+        [string]$ServerInstanceUser,
+        [securestring]$ServerInstancePassword
+    )
+    try {
+        if (!($ServerInstanceUser) -or !($ServerInstancePassword)) {
+            $exists = Invoke-SQLCmd -ServerInstance $ServerInstance -Query "SELECT name FROM syslogins WHERE name = '$LoginName'"
+        }
+        else {
+            $SQLCredentials = New-Object -TypeName System.Management.Automation.PSCredential ($ServerInstanceUser, $ServerInstancePassword)
+            $exists = Invoke-SQLCmd -ServerInstance $ServerInstance -Query "SELECT name FROM syslogins WHERE name = '$LoginName'" -Credential $SQLCredentials
+        }
+    }   
+    catch {
+        Fail-Json -obj $result -message $($_.exception.message)
+    }
+
+    if ($exists) {
+        return $true
+    }
+
+    return $false
+}
+
+Function Add-SQLLogin {
+
+    Param (
+        [string]$LoginName,
+        [string]$Password,
+        [string]$LoginType,
+        [string]$DefaultDatabase,
+        [string]$DefaultLanguage,
+        [bool]$CheckExpiration,
+        [bool]$CheckPolicy,
+        [string]$ServerInstance,
+        [string]$ServerInstanceUser,
+        [securestring]$ServerInstancePassword,
+        [bool]$CheckMode
+    )
+
+    $Expiration = 'CHECK_EXPIRATION=OFF'
+    $Policy = 'CHECK_POLICY=OFF'
+
+    if ($CheckExpiration) {
+        $Expiration = 'CHECK_EXPIRATION=ON'
+    }
+    if ($CheckPolicy) {
+        $Policy = 'CHECK_POLICY=ON'
+    }
+
+    if ($LoginType -eq 'sql') {
+        $QUERY = "CREATE LOGIN $LoginName WITH PASSWORD=N'$Password', DEFAULT_DATABASE=$DefaultDatabase, DEFAULT_LANGUAGE=$DefaultLanguage, $Expiration, $Policy"
+    }
+    else {
+        if ($LoginName -match "\\") {
+            $LoginName = $LoginName.Replace("\\", "\")
+        }
+        $QUERY = "CREATE LOGIN `"$LoginName`" FROM WINDOWS WITH DEFAULT_DATABASE=$DefaultDatabase, DEFAULT_LANGUAGE=$DefaultLanguage"
+    }
+    
+    $exists = Test-SQLUser -LoginName $LoginName `
+        -ServerInstance $ServerInstance `
+        -ServerInstanceUser $ServerInstanceUser `
+        -ServerInstancePassword $ServerInstancePassword
+    
+    if (!($exists)) {
+        try {
+            if (!($CheckMode)) {
+                if (!($ServerInstanceUser) -or !($ServerInstancePassword)) {
+                    Invoke-SQLCmd -ServerInstance $ServerInstance -Query $QUERY
```

Comment:
We should be using the `*-SqlLogin` cmdlets for these operations and not crafting the queries ourselves. They can deal with all the escaping issues  and handle whatever checks are necessary.